mm2src/coins/hd_wallet_storage/wasm_storage.rs

```diff
@@ -0,0 +1,357 @@
+            .open_cursor("wallet_id")
+            .await?
+            .only("coin", wallet_id.coin)?
+            .only("mm2_rmd160", wallet_id.mm2_rmd160)?
+            .only("hd_wallet_rmd160", wallet_id.hd_wallet_rmd160)?
+            .collect()
+            .await?
+            .into_iter()
+            .map(|(item_id, _item)| item_id);
+
+        // Delete accounts by `item_ids`.
+        for account_item_id in item_ids {
+            table.delete_item(account_item_id).await?;
+        }
+
+        Ok(())
+    }
+}
+
+impl HDWalletIndexedDbStorage {
+    fn get_shared_db(&self) -> HDWalletStorageResult<SharedDb<HDWalletDb>> {
+        self.db
+            .upgrade()
+            .or_mm_err(|| HDWalletStorageError::Internal("'HDWalletIndexedDbStorage::db' doesn't exist".to_owned()))
+    }
+
+    async fn lock_db(db: &SharedDb<HDWalletDb>) -> HDWalletStorageResult<HDWalletDbLocked<'_>> {
+        db.get_or_initialize().await.mm_err(HDWalletStorageError::from)
+    }
+
+    async fn find_account<'a>(
+        table: &DbTable<'a, HDAccountTable>,
+        wallet_id: HDWalletId,
+        account_id: u32,
+    ) -> HDWalletStorageResult<Option<(ItemId, HDAccountTable)>> {
+        // Use the cursor to find an item with the specified `wallet_id` and `account_id`.
+        let accounts = table
+            .open_cursor("wallet_account_id")
+            .await?
+            .only("coin", wallet_id.coin)?
+            .only("mm2_rmd160", wallet_id.mm2_rmd160)?
+            .only("hd_wallet_rmd160", wallet_id.hd_wallet_rmd160)?
+            .only("account_id", account_id)?
+            .collect()
+            .await?;
+
+        if accounts.len() > 1 {
+            let error = DbTransactionError::MultipleItemsByUniqueIndex {
+                index: "wallet_account_id".to_owned(),
+                got_items: accounts.len(),
+            };
+            return MmError::err(HDWalletStorageError::ErrorLoading(error.to_string()));
+        }
+        Ok(accounts.into_iter().next())
+    }
+
+    async fn update_account<F>(&self, wallet_id: HDWalletId, account_id: u32, f: F) -> HDWalletStorageResult<()>
+    where
+        F: FnOnce(&mut HDAccountTable),
+    {
+        let shared_db = self.get_shared_db()?;
+        let locked_db = Self::lock_db(&shared_db).await?;
+
+        let transaction = locked_db.inner.transaction().await?;
+        let table = transaction.table::<HDAccountTable>().await?;
+
+        let (account_item_id, mut account) = Self::find_account(&table, wallet_id.clone(), account_id)
+            .await?
+            .or_mm_err(|| HDWalletStorageError::HDAccountNotFound { wallet_id, account_id })?;
+
+        // Apply `f` to `account` and upload the changes to the storage.
+        f(&mut account);
+        table
+            .replace_item(account_item_id, &account)
+            .await
+            .map(|_| ())
+            .mm_err(HDWalletStorageError::from)
+    }
+}
+
+/// This function is used in `hd_wallet_storage::tests`.
```

Comment:
Do we miss a `#[cfg(test)]` here ?

Reply:
Unfortunately, I couldn't figure out how to run tests in WASM from the module with `#[cfg(test)]`.
According to [docs](https://rustwasm.github.io/wasm-bindgen/wasm-bindgen-test/usage.html#write-some-tests):
> One other difference is that the tests must be in the root of the crate, or within a pub mod. Putting them inside a private module will not work.

I tested it a bit, and in our repo it works if the `#[cfg(test)]` attribute belongs to a module within the root crate, in particular, to a `pub mod` within `mm2_tests`, `docker_tests`. It doesn't work for pub modules within `coin`, `common` crates etc.